src/DynamoCore/Core/CustomNodeManager.cs

```diff
@@ -135,34 +135,97 @@ protected virtual void OnCustomNodeRemoved(Guid functionId)
         public Function CreateCustomNodeInstance(
             Guid id, string name = null, bool isTestMode = false)
         {
-            CustomNodeWorkspaceModel workspace;
-            CustomNodeDefinition def;
-            CustomNodeInfo info;
+            CustomNodeDefinition def = null;
+            CustomNodeInfo info = null;
+            TryGetCustomNodeData(id, name, isTestMode, out def, out info);
+
+            return CreateCustomNodeInstance(id, name, isTestMode, def, info);
+        }
+
+        /// <summary>
+        ///     Attempts to get custom node info and definition data.
+        /// </summary>
+        /// <param name="id">Identifier referring to a custom node definition.</param>
+        /// <param name="name">
+        ///     Name for the custom node to be instantiated, used for error recovery if
+        ///     the given id could not be found.
+        /// </param>
+        /// <param name="isTestMode">
+        ///     Flag specifying whether or not this should operate in "test mode".
+        /// </param>
+        /// <param name="def">
+        ///     Custom node definition data
+        /// </param>
+        /// <param name="info">
+        ///     Custom node information data
+        /// </param>
+        public bool TryGetCustomNodeData(
+            Guid id, 
+            string name,
+            bool isTestMode, 
+            out CustomNodeDefinition def, 
+            out CustomNodeInfo info)
+        {
+            def = null;
+            info = null;
+
             // Try to get the definition, initializing the custom node if necessary
             if (TryGetFunctionDefinition(id, isTestMode, out def))
             {
                 // Got the definition, proceed as planned.
                 info = NodeInfos[id];
+                return true;
             }
-            else
+
+            // Couldn't get the workspace with the given ID, try a name lookup instead.
+            if (name != null && !TryGetNodeInfo(name, out info))
+                return false;
+
+            // Try to get the definition using the function ID, initializing the custom node if necessary
+            if (info != null && TryGetFunctionDefinition(info.FunctionId, isTestMode, out def))
```

Comment:
@ColinDayOrg It seems `TryGetFunctionDefinition()` and the code path underneath are still based on xml doc, do you see todos while developing? Just curious.

Reply:
It does appear that there are multiple issues with custom nodes around XML versus JSON (some of which I believe Sriram just addressed), but it seemed somewhat out of scope to me for the current task. I would expect that once the dummy node code is merged that there would be a fair amount of testing that will expose some issues that would be logged as bugs and addressed in the future. If you think it would be more appropriate I can add "TODO" comments.

Reply:
That's fine. I was just curious anything you noticed as known issues.